### ISSUE TITLE
couchpotato + rtorrent confusion about folder path

### ISSUE THREAD

**Issue by SJ50** [clinton-hall/nzbToMedia#788]
Hi,
I am little confused after reading wiki on how to configure couchpotato and autoProcessMedia.cfg file.

I want to set up.
rtorrent download movie in "/media/hdd1/rtorrent/download/movie" I want that downloaded file to move to "/media/hdd2/media/movie" and create symlink to original download folder so I can seed forever. At this moment I have 50 files being seeded and all of them are symlinks which I created manually. 
Now I want couchpotato to take care this and I am confuse. I have created .ignore file so couchpotato do no process existing file and I have took backup everything in case anything goes wrong.

as per this link https://github.com/clinton-hall/nzbToMedia/wiki/couchpotato
rtorrent download folder and from folder should be different. 
so, I set my my rtorrent download folder as "/media/hdd1/rtorrent/download/movie"
my TO folder for renamer is "/media/hdd2/media/movie" now which path should I give for From folder in renamer setting?
In autoProcessMedia.cfg there in Couchpotato section there is option to setup  watch_dir = "" should I give any path here?

as of now I gave some fake path to From folder in renamer setting and I left watch_dir path empty.

Thank you

**Comment by clinton-hall** [comment]
Keep Renamer "From" as a "fake path.
for watch_dir, I recommend leaving this blank. It is only useful for manual.

In autoProcessMedia.cfg
```
[Torrent]
    useLink = sym
    outputDirectory = /media/hdd1/rtorrent/processed
```
now, in CouchPotato settings, renamer, `run_every = 0`, `Default File Action = move`, `Torrent File Action = move` and restart CouchPotato.

the flow is this:
TorrentToMedia is called by rtorrent and creates a symlink (to each file) at /media/hdd1/rtorrent/processed/movie/moviename/
TorrentToMedia tells couchpotato to scan /media/hdd1/rtorrent/processed/movie/moviename/
and CouchPotato then moves the symlink files to /media/hdd2/media/movie/moviename/

**Comment by frriction** [comment]
This method keeps original file in /media/hdd1/rtorrent/download/movie which is rtorrent download folder I want to move original file to destination folder and create sym link in place of original file.
space of hdd1 is limited.

**Comment by frriction** [comment]
I tried doing this now and it worked but risky it may delete other movie.

````    
[CouchPotato]
...
        Torrent_NoLink = 1
        method = renamer
        delete_failed = 0
        wait_for = 10
        extract = 0
...
        watch_dir = ""
....
[Torrent]
    ###### clientAgent - Supported clients: utorrent, transmission, deluge, rtorrent, vuze, other
    clientAgent = rtorrent
    ###### useLink - Set to hard for physical links, sym for symbolic links, move to move, and no to not use links (copy)
    useLink = copy
    ###### outputDirectory - Default output directory (categories will be appended as sub directory to outputDirectory)
    outputDirectory = /media/pendrive/rtorrent/download/
    ###### Enter the default path to your default download directory (non-category downloads). this directory is protected by safe_mode.
    default_downloadDirectory = /media/pendrive/rtorrent/download/
  ````  

In Couchpotato 
Renamer setting as follows 
`Run Every = 0` 
`To = /media/hdd2/media/movie` 
`From = /tmp` 
`Force Every = 0`
`Torrent File Action = Link`

added .ignore file for existing symlinks in my rtorrent download folder
downloaded a file which executed script and it worked. 
CP copied the file from rtorrent download folder to destination folder and created symlink in download folder which points to the destination folder. 

BUG
I forgot to create .ignore file for one of the symlink which was in separate folder even that existing file got post processed; Now CP has tendency to delete files in destination folder if it post process again same movie 
` [tato.core.plugins.renamer] Same quality release already exists for xxxxxxxx, with quality 720p. Assuming repack.`
which deleted that xxxxxxxx file in my destinatino folder. 
More over it did not created .ignore file for recently postprocessed file, I will download more file and see it CP tries to delete this file and if yes than what can I do to prevent it.

**Comment by frriction** [comment]
I have change from path in couchpotato to "/media/pendrive/rtorrent/download/movie" and it creates .ignore file and everything is fine. 
However there is one issue CP scans From folder when it starts (power failure or other reason) and at that time if you have any unfinished torrent running than CP will post process that file and you end up with unfinished file. 
I am thinking to use Autotools to prevent that to happening but in rtorrent config file line which calls for nzbtomedia interferes with automove. 
is there any way to finish automove function than call for nzbtomedia script?
or is there any other way which updates path of torrent to new path and call nzbtomedia script?

**Comment by clinton-hall** [comment]
Does rtorrent allow incomplete downloads to be downloaded to one location and then only moved to another location on torrent complete? Or is that what auto tools does?
I'm not familiar with auto tools or how to use this.

**Comment by frriction** [comment]
Yes, it allows to download at one location and move it other location upon completion of torrent and update update path appropriately in rtorrent so it keeps seeding torrent. 

for example if I have base path /home/download/movie and I have labeled that download as movie, and in autotool If I set base path as /home/complete, upon completion of torrent autotool will move torrent to /home/complete/movie

here is a screenshot  ![Image of autotools] (http://i.imgur.com/E4FTNvx.png)

[link to its wiki page!] (https://github.com/Novik/ruTorrent/wiki/PluginAutotools)
wiki page is little outdated but gives idea what it does. 

In rtorrent config file we have to add
 ```system.method.set_key = event.download.finished,TorrentToMedia,"execute={/usr/bin/python,/path/to/nzbToMedia/TorrentToMedia.py,\"$d.get_base_path=\",\"$d.get_name=\",\"$d.get_custom1=\",\"$d.get_hash=\"}"
``` which prevents autotools to work.

**Comment by clinton-hall** [comment]
I don't know if symlink of renamed files can be easily achieved via these scripts... I really don't know what else to suggest here. Sym/hard linking from download to renamed is easy, but linking back to original presents the issues you have highlighted. 

If you can identify a process change, I can see if I can support it, but I can't see exactly what needs to be changed to make this happen.

**Comment by frriction** [comment]
Well, I can see three options here. 
Option 1) This may be you can support 
 In autoProcessMedia.cfg if I use
"
[Torrent]
    useLink = move
 outputDirectory=/media/hdd1/rtorrent/processed
"
Than CP moves the file and creates symlink at /media/hdd1/rtorrent/processed
Now if some automated way if I can place this symlink to original download folder. This way torrent keeps seeding and CP (at the time of CP start) will not process half downloaded file. 
Option 2) I find some way to delay calling of nzbtomedia script untill autool finished moving. 
Option 3) If CP never tries to post-process not even at start of CP. I have already open issue there (almost 6 month back)  but I do not know when this gets resolved. 

With SickRage moving and symlink back works perfectly. 
At this momemt I wish I knew atleast some basic script writing. 
Thank you for all the support you provided.

**Comment by frriction** [comment]
I got it working.

Here is what I did.
Disabled automove feature from autotools. 
now I modified .rtorrent.rc (rtorrent config file) and added 
```
system.method.insert=d.get_finished_dir,simple,"cat=/home/pi/rtorrent/complete/,$d.get_custom1="
```

above line defines destination directory based on label.
d.get_finished_dir represent destination and d.get_custom1 represent label.
so above line appends directory with label name.


```
system.method.set_key = event.download.finished,move_complete,"d.set_directory=$d.get_finished_dir=;execute={/home/pi/rtorrent.sh,$d.get_base_path=,$d.get_name=,$d.get_custom1=,$d.get_hash=,$d.get_finished_dir=}"
```
above line change path of the torrent to destination directory and calls for bash script which calls  nzbtomedia to post process downloaded torrent.

here is bash script 
```
#!/bin/bash
LOG="/home/pi/log"
#echo "d.get_base_path = $1" >> "$LOG"
#echo "d.get_name = $2" >> "$LOG"
#echo "d.get_custom1 = $3" >> "$LOG"
#echo "d.get_hash = $4" >> "$LOG"
#echo "d.get_finished_dir = $5" >> "$LOG"
#echo "------------------------" >> "$LOG"
mkdir -p "$5"
mv "$1" "$5"
/usr/bin/python2 /srv/http/nzbToMedia/TorrentToMedia.py "$5" "$2" "$3" "$4"
```

above script creates necessary directory and move content of torrent to destination directory than calls TorrentToMedia python script with necessary parameters. 

I am not expert, just a advanced noob who can googled. It took me 1 hour to write that bash script so you can measure my level here. 
If any one find any better solution or suggestion please comment so I can learn.

**Comment by clinton-hall** [comment]
Good job. That certainly looks like the right process.

I didn't suggest an Sh script to move then call torrenttomedia because I assumed this would break seeding... I gather this is what the 2 edits to rtorrent.rc do? They define the new dir, call the script, and tell rtorrent to look for files in the new dir?

**Comment by frriction** [comment]
first line defines new path for downloaded torrent while  this part `d.set_directory=$d.get_finished_dir=` in second edit of rtorrent.rc, updates path of the torrent downloaded and changed it to "complete" directory at this moment download content are in download folder but path points to complete so I added move command in Sh script.
There is way to add mv command in rtorrent.rc itself but before mv command finishes its operation nzbtomedia script gets called.
I have tested this method with 9 different torrent with same label though every thing works so finger crossed that it works for other label too.
